Improve db population (#567)

Didi-121 opened the pull request:
- Added a flag (--regenerate-db) to run.sh that runs regenerate_db.sh
- This script deletes current dumps, generate new dumps based on the jsons, truncate tables at DB and insert the new dumps
- Auto complete modified for this flag
- Eliminated redundancies on create_sql_dump.py
- Conserved dummy data for each table (data used at robocup 2025)

Test:
Content of a table before using --regenerate-db
<img width="1567" height="232" alt="image" src="https://github.com/user-attachments/assets/08c0aedc-a608-429d-b4c4-a22243df30d6" />

Logs of run.sh (using --regenerate-db)  after changing data from knowledge bases (jsons)
<img width="1600" height="717" alt="image" src="https://github.com/user-attachments/assets/745cb141-3a8d-4a97-bbbc-0121f90a16be" />

Content of table after the task
<img width="1544" height="370" alt="image" src="https://github.com/user-attachments/assets/4e9d9321-0e07-4d63-9818-966f0a02cfde" />

Didi-121 committed:
Improve db population (#567)
- Added a flag to run.sh that runs regenerate_db.sh
- This script deletes current dumps, generate new dumps based on the jsons, truncate tables at DB and insert the new dumps
- Auto complete modified for this flag
- ELiminated redundancies on create_Sql_dump.py
- Conserved dummy data for each table (data used at robocup 2025)

GilMM27 commented on docker/hri/scripts/regenerate_db.sh:
if this runs inside hri-ros this seems to be written for running outside

Didi-121 replied:
Indeed the script is written for running outside for easier handling when accessing the generated dumps and executing queries to postgres container. Also hri-ros is just needed for creating the new dumps.

GilMM27 commented on docker/hri/scripts/regenerate_db.sh:
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+ENV_TYPE=${1:-cpu}
```
let's not default because that might attempt building the wrong images for users

GilMM27 commented on docker/hri/scripts/regenerate_db.sh:
```diff
@@ -0,0 +1,78 @@
+    break
+  fi
+  sleep 1
+done
```
this seems like a complicated way of running docker ps and checking if they are up

GilMM27 commented on docker/hri/scripts/regenerate_db.sh:
```diff
@@ -0,0 +1,78 @@
+
+# Extract table as the first segment after the prefix
+for file in "$DUMPS_DIR"/04-*.sql; do
+  [ -f "$file" ] || continue
```
shouldn't the for take care of this?

GilMM27 commented on docker/hri/scripts/regenerate_db.sh:
```diff
@@ -0,0 +1,78 @@
+  # Verify table exists in public schema
+  exists=$(docker compose -f "$COMPOSE" exec -T postgres psql -U rbrgs -d postgres -tAc "SELECT to_regclass('public.\"$tbl\"');" | tr -d '[:space:]' || true)
+  if [ -z "$exists" ] || [ "$exists" = "(null)" ]; then
+    echo "  Table '$tbl' not found in database, skipping $base." >&2
```
could use \t instead of the blank spaces

GilMM27 reviewed (changes requested):
good work

Didi-121 commented on frida_constants/data/areas.json:
Yes, this files were expected by create_sql_dump.py. I added this files based on the expected output (the sql dump that was already present at the repo)

GerardoFJ replied:
This files are already created at
https://github.com/RoBorregos/home2/tree/main/frida_constants/map_areas

Didi-121 commented on docker/hri/scripts/regenerate_db.sh:
```diff
@@ -0,0 +1,78 @@
+    echo "  Error: failed to load $base into $tbl" >&2
+  else
+    echo "  Loaded $base successfully."
+  fi
```
Here the dumps are loaded to the DB.

Didi-121 committed:
fix/correct regenerate_db flag handling

-Changed the behavior of the argument from a task to an optional setup process
-The tables are now detected dynamically and only the tables with an existent sql dump are regenerated
-Minor changes as: deletion of commented code, unesessary checks, defaulting, etc

GilMM27 commented on docker/hri/run.sh:
```diff
@@ -110,7 +119,6 @@ IGNORE_PACKAGES="--packages-ignore frida_interfaces frida_constants xarm_msgs"
 SOURCE_ROS="source /opt/ros/humble/setup.bash"
 PACKAGES="speech nlp embeddings"
 PROFILES=()
-RUN=""
```
why'd you delete it?

Didi-121 replied:
by mistake, I just restored it

GilMM27 reviewed (commented):
Well done, left a few comments. Also good rule of thumb for adding comments in code is to add a blank space after the `#` and start with a capital letter ;) super nit tho jaja

Didi-121 committed:
fix: context_areas.json restored

Didi-121 commented:
The requested changes were made. Also the file context_areas.json has been restored because the dump generator requires it to function properly. It was originally added on July 8, 2025 (commit b3eaeff0), but was mistakenly deleted during a cleanup on July 28, 2025 (commit 9b9b34f5 "Deleted trash areas"), where 19 area-related files were removed. Unlike the other deleted files, context_areas.json contains the context necessary for the current areas file.

The following images provide evidence of the test on the Orin:
-The commit/version of the repository is shown.
-The database table "items" is displayed.
-The JSON file that generates the dump for items table is edited.
-The script is executed.
-Finally, the same table is shown again, reflecting the modifications.
<img width="1830" height="1124" alt="image" src="https://github.com/user-attachments/assets/8d63a9a7-d5da-437f-8ebd-7893eeb8384d" />
<img width="1829" height="1097" alt="image" src="https://github.com/user-attachments/assets/200eab76-7a02-4836-b88e-133c3972fc1e" />

GilMM27 reviewed (approved):
great work, let's work on the follow ups mentioned in the conversation next
